bundles/statistics/statsgrid2016/components/Diagram.js

```diff
@@ -248,14 +248,22 @@ Oskari.clazz.define('Oskari.statistics.statsgrid.Diagram', function (service, lo
      * @return colors[] containing colors
      */
     getColorScale: function () {
-        /*
+        var me = this;
         var stateService = this.service.getStateService();
         var activeIndicator = stateService.getActiveIndicator();
         var classificationOpts = stateService.getClassificationOpts(activeIndicator.hash);
+        var currentRegionset = stateService.getRegionset();
+        var classification = {};
+        this.service.getIndicatorData(activeIndicator.datasource, activeIndicator.indicator, this.service.getIndicatorData(activeIndicator.datasource,
```

Comment:
The third param is this.service.getIndicatorData(). That seems erronous. Also getIndicatorData() is async although in practice it's probably always sync at this point. Would it be possible to pass the data as param to this method?